tests/unit/test_coresubset.py

```diff
@@ -534,6 +880,213 @@ def test_random_sample_with_replacement(self) -> None:
+        # this test
+        random_sample = coreax.coresubset.RandomSample(
+            random_key=self.random_key,
+            refine_method=coreax.util.InvalidKernel,
+        )
+
+        # The fit method should just randomly select points - so we expect to be able to
+        # call fit without any errors being raised
+        random_sample.fit(
+            original_data=self.data_obj,
+            strategy=coreax.reduction.SizeReduce(self.coreset_size),
+        )
+
+        # Now, if we refine the coreset generated during the call to fit, we will use
+        # the refine method, so expect an error to be raised
+        with self.assertRaises(AttributeError) as error_raised:
+            random_sample.refine()
+
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "type object 'InvalidKernel' has no attribute 'refine'",
+        )
+
+    def test_random_sample_fit_zero_coreset_size(self):
+        """
+        Test how random sample performs when given a zero value of coreset_size.
+        """
+        # Define a kernel herding object
+        random_sample = coreax.coresubset.RandomSample(random_key=self.random_key)
+
+        # Call the fit method with a coreset size of 0 - this should just sample zero
+        # points at random, so not raise any issues
+        random_sample.fit(
+            original_data=self.data_obj,
+            strategy=coreax.reduction.SizeReduce(coreset_size=0),
+        )
+        self.assertEqual(len(random_sample.coreset_indices), 0)
+        self.assertEqual(len(random_sample.coreset), 0)
+
+    def test_random_sample_fit_negative_coreset_size(self):
+        """
+        Test how random sample performs when given a negative value of coreset_size.
+        """
+        # Define a kernel herding object
+        random_sample = coreax.coresubset.RandomSample(
+            random_key=self.random_key,
+        )
+
+        # Call the fit method with a negative coreset size - this should try to run a
+        # JAX loop with start and end points being the same, and index an empty array,
+        # so raise a value error
+        with self.assertRaises(ValueError) as error_raised:
+            random_sample.fit(
+                original_data=self.data_obj,
+                strategy=coreax.reduction.SizeReduce(coreset_size=-2),
+            )
+        self.assertEqual(
+            error_raised.exception.args[0],
+            "coreset_size must be a positive integer",
+        )
+
+    def test_random_sample_fit_float_coreset_size(self):
+        """
+        Test how random sample performs when given a float value of coreset_size.
+        """
+        # Define a kernel herding object
+        random_sample = coreax.coresubset.RandomSample(
+            random_key=self.random_key,
+        )
+
+        # Call the fit method with a float value for coreset size - this should error
+        # hen trying to define an integer number of samples
```

Comment:
Missing a letter. Assuming it's meant to be 'when'
```suggestion
        # when trying to define an integer number of samples
```

Reply:
Thanks for catching - updated